lib/Wrapper/OptimizingScheduler.cpp

```diff
@@ -237,8 +242,10 @@ void ScheduleDAGOptSched::SetupLLVMDag() {
   // Finalize live-in
   RPTracker.closeTop();
 
-  Topo.InitDAGTopologicalSorting();
-  postprocessDAG();
+  if (EnableMutations) {
+    Topo.InitDAGTopologicalSorting();
+    postprocessDAG();
+  }
```

Comment:
I don't quite understand why these should only be run if mutations are enabled.

Reply:
I am not exactly too sure about `Topo.InitDAGTopologicalSorting()` myself but I believe the information is used when the mutation adds the edges. `addMutation()` only adds the desired mutations to a list and does not apply it until `postprocessDAG();` is called.

Reply:
I'm not sure about `Topo.InitDAGTopologicalSorting()` ( https://llvm.org/doxygen/classllvm_1_1ScheduleDAGTopologicalSort.html#a44c5faa549f250a26b1303eb1a3ebd47 ) either. I don't know if we use the topological sort anywhere. If this has been tested and it works, then I think it should be fine.